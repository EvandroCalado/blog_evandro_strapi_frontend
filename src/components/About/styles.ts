import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.frameSizes['4xl']};
    margin: 0 auto;
    padding: 0 ${theme.spacings.md};
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
    gap: ${theme.spacings.xl};

    ${Title} {
      font-size: ${theme.font.sizes['4xl']};
    }
`}
`;

export const Description = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.md};

    &::first-letter {
      font-size: ${theme.font.sizes.xl};
      font-weight: bold;
    }
`}
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css`

    & > img {
      width: 35rem;
      height: 35rem;
      overflow: hidden;
      border-radius: ${theme.spacings.md};
      object-fit: cover;
    }
`}
`;
