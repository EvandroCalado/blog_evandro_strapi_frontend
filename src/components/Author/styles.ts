import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.frameSizes['4xl']};
    display: flex;
    align-items: center;
    margin: ${theme.spacings.xl} auto;

    @media screen and ${theme.media.lteOrEqSmall} {
      flex-direction: column;
      width: 30rem;
    }
`}
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css`
    position: relative;
    width: 22rem;
    height: 17rem;
    border-radius: ${theme.spacings.sm};
    overflow: hidden;
    margin-right: ${theme.spacings.md};

    & img {
      object-fit: cover;
    }

    @media screen and ${theme.media.lteOrEqSmall} {
      width: 15rem;
      margin-bottom: ${theme.spacings.lg};
    }
`}
`;

export const ContentContainer = styled.div`
  ${({ theme }) => css`
    width: 100%;

    @media screen and ${theme.media.lteOrEqSmall} {
      display: flex;
      flex-direction: column;
      align-items: center;
      justify-content: center;
    }
`}
`;

export const Content = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.sm} 0;
    overflow: hidden;
    display: -webkit-box;
    -webkit-box-orient: vertical;
    -webkit-line-clamp: 3;
`}
`;
