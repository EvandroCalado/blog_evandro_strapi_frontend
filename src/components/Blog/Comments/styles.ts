import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.frameSizes['4xl']};
    margin: 0 auto;

    iframe {
      color: ${theme.colors.text};
    }
`}
`;
