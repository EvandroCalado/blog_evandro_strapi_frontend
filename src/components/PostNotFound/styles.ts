import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    width: 100%;
    display: flex;
    height: 54.6rem;
    align-items: center;
    justify-content: center;

    @media screen and ${theme.media.lteOrEqSmall} {
      height: 40rem;
    }
`}
`;
