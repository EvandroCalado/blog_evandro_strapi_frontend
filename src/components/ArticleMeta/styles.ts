import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.frameSizes['7xl']};
    margin: auto;
    padding: 0 ${theme.spacings.md};
    display: flex;
    align-items: center;
    flex-direction: column;
    gap: ${theme.spacings.xl};

    & > span > a {
      font-weight: bold;
      transition: ${theme.transitions.faster};
      margin-left: ${theme.spacings.sm};

      &:hover {
        text-decoration: underline;
      }
    }

    ${Title} {
      width: 100%;
      max-width: ${theme.frameSizes['4xl']};
      text-align: center;
    }
`}
`;

export const Image = styled.div`
  ${({ theme }) => css`
    position: relative;
    width: 100%;
    height: 40rem;
    object-fit: cover;
    border-radius: ${theme.spacings.sm};
    overflow: hidden;
  `}
`;

export const ArticleContainer = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.frameSizes['4xl']};
    display: flex;
    align-items: center;
    justify-content: space-between;

    @media screen and ${theme.media.lteOrEqSmall} {
      flex-direction: column;
    }
`}
`;

export const DateContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: end;
    justify-content: center;
    gap: ${theme.spacings.lg};
`}
`;
