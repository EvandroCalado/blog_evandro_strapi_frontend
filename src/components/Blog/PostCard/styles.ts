import styled, { css } from 'styled-components';
import { Title } from '../../Shared/Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    flex-direction: column;
    padding: ${theme.spacings.lg} ${theme.spacings.md};

    & > a {
      width: 100%;
      height: 20rem;
      margin-bottom: ${theme.spacings.md};
    }
`}
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css`
    position: relative;
    width: 100%;
    height: 100%;
    margin-bottom: ${theme.spacings.md};

    & img {
      object-fit: cover;
      border-radius: ${theme.spacings.sm};
    }
`}
`;

export const ContentContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    gap: ${theme.spacings.md};

    & > a {
      transition: ${theme.transitions.faster};

      &:hover {
        text-decoration: underline;
        filter: invert(.2);
      }
    }

    ${Title} {
      overflow: hidden;
      display: -webkit-box;
      -webkit-box-orient: vertical;
      -webkit-line-clamp: 2;
    }
`}
`;

export const Content = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.xs} 0;
    overflow: hidden;
    display: -webkit-box;
    -webkit-box-orient: vertical;
    -webkit-line-clamp: 4;
`}
`;

export const DateContainer = styled.div`
    display: flex;
    align-items: center;
    justify-content: space-between;
`;

export const Date = styled.span`
`;
