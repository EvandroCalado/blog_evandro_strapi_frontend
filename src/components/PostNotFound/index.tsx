import Heading from '../Heading';
import * as Styled from './styles';

export default function PostNotFound() {
  return (
    <Styled.Container>
      <Heading>Post n√£o encontrado !</Heading>
    </Styled.Container>
  );
}
