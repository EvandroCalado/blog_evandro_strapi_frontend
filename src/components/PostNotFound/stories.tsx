import { Meta } from '@storybook/react';
import PostNotFound from '.';

export default {
  title: 'Components/PostNotFound',
  component: PostNotFound,
  parameters: {
    layout: 'fullscreen',
  },
} as Meta;

export const Default = () => <PostNotFound />;
