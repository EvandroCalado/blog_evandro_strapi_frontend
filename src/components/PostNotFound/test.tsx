import '@testing-library/jest-dom';
import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import PostNotFound from './';

describe('<PostNotFound />', () => {
  it('should render with default values', () => {
    renderTheme(<PostNotFound />);
    expect(
      screen.getByRole('heading', { name: 'Post nÃ£o encontrado !' }),
    ).toBeInTheDocument();
  });

  it('should render a snapshot', () => {
    const { container } = renderTheme(<PostNotFound />);
    expect(container).toMatchSnapshot();
  });
});
