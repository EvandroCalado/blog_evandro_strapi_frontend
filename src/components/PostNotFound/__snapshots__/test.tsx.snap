// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostNotFound /> should render a snapshot 1`] = `
.c1 {
  text-transform: none;
}

.c0 {
  width: 100%;
  display: flex;
  height: 54.6rem;
  align-items: center;
  justify-content: center;
}

@media screen and (max-width: 576px) {
  .c0 {
    height: 40rem;
  }
}

<div>
  <div
    class="c0"
  >
    <h1
      class="c1"
    >
      Post n√£o encontrado !
    </h1>
  </div>
</div>
`;
